Configure.ps1

```diff
@@ -84,25 +85,32 @@ so that they are consistent with the Applications parameters
     # $tenantId is the Active Directory Tenant. This is a GUID which represents the "Directory ID" of the AzureAD tenant 
     # into which you want to create the apps. Look it up in the Azure portal in the "Properties" of the Azure AD. 
 
-    # Login to Azure PowerShell (interactive: you'll need to sign-in with creds enabling your to create apps in the tenant)
-    $creds = Connect-AzureAD -TenantId $tenantId
+    # Login to Azure PowerShell (interactive if credentials are not already provided: 
+    # you'll need to sign-in with creds enabling your to create apps in the tenant)
+    if (!$Credential)
+    {
+        $creds = Connect-AzureAD -TenantId $tenantId
+    }
+    else
+    {
+        if (!$TenantId)
+        {
+            $creds = Connect-AzureAD -Credential $Credential
+        }
+        else
+        {
+            $creds = Connect-AzureAD -TenantId $tenantId -Credential $Credential
+        }
+    }
+
     if (!$tenantId)
     {
         $tenantId = $creds.Tenant.Id
     }
     $tenant = Get-AzureADTenantDetail
     $tenantName =  $tenant.VerifiedDomains[0].Name
 
-    # Variables for the registration of the AAD application for the Web API Service
```

Comment:
These variables were factored out in the Parameter.ps1 script. They are used both by this script (Configure.ps1) and by the cleanup script (Cleanup.ps1). This is also easier for customers to parametrize their application as the Parameters.ps1 file contains only the parameters, no PowerShell code.